### PR TITLE
CA-548 release connection at end of ldapSearchStream, don't close

### PR DESCRIPTION
**Pull request by dvoet**
Ticket: https://broadworkbench.atlassian.net/browse/CA-548
during profiling I saw that ldap threads were being created and killed in great numbers despite connection pooling. I discovered that `close` on a connection really closes it even if it is part of a pool. Generally we let unboundid handle getting and release connections from the pool but there is this one place where we do it ourselves.

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [ ] I've updated the [FISMA documentation](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

### CODE REVIEW

**Review by marctalbott — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala [hunk 1]
```diff
@@ -36,27 +36,32 @@ trait LdapSupport extends DirectorySubjectNameSupport {
   protected def ldapSearchStream[T](baseDn: String, searchScope: SearchScope, filters: Filter*)(unmarshaller: Entry => T): Stream[T] =
     filters.flatMap { filter =>
       val search = new SearchRequest(baseDn, searchScope, filter)
-      val entrySource = new LDAPEntrySource(ldapConnectionPool.getConnection, search, true)
-      ldapEntrySourceStream(entrySource)(unmarshaller)
+      val connection = ldapConnectionPool.getConnection
+      val entrySource = new LDAPEntrySource(connection, search, false)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala]: notice that the last param switched from true to false

**Review by davidangb — APPROVED** [review]
:+1:

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala (hunk 1 above)
Comment (helgridly) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala]: nit: can you put the bool variable name in the call so it's clear what it does

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala [hunk 2]
```diff
@@ -36,27 +36,32 @@ trait LdapSupport extends DirectorySubjectNameSupport {
...
+      ldapEntrySourceStream(entrySource, connection)(unmarshaller)
     }.toStream
 
   // this is the magic recursive stream generator
-  private def ldapEntrySourceStream[T](entrySource: LDAPEntrySource)(unmarshaller: Entry => T): Stream[T] =
+  private def ldapEntrySourceStream[T](entrySource: LDAPEntrySource, connection: LDAPConnection)(unmarshaller: Entry => T): Stream[T] =
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala:48]: so you create a single connection in `ldapSearchStream` and pass it through here but where does it actually get used to do work? all i see is it getting released. or is that related to the mystery bool?

**Review by helgridly — COMMENTED** [review]
honestly i don't understand what's happening here

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala (hunk 2 above)
Comment (dvoet) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala:48]: line 40 is where it actually does the query. The rest of it is reading off the `LDAPEntrySource` until there are no more entries. In the old code the connection was automatically closed by `LDAPEntrySource`. The new code tells `LDAPEntrySource` not to do that then releases the connection when there are no more entries.

Comment (helgridly) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/util/LdapSupport.scala:48]: thanks!

**Review by helgridly — APPROVED** [review]

**Comment by dvoet** [comment]
jenkins retest
